File: test_app/integration_tests/android_mocks/src/main/java/com/voxeet/sdk/services/VideoPresentationService.java
package com.voxeet.sdk.services;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.voxeet.promise.Promise;
import com.voxeet.sdk.services.presentation.PresentationState;
import com.voxeet.sdk.services.presentation.video.VideoPresentation;

import java.util.ArrayList;
import java.util.List;

/**
 * VideoPresentationService mock
 */
public class VideoPresentationService {
    protected List<VideoPresentation> presentations = new ArrayList<>();
    public PresentationState currentState = PresentationState.STOP;
    public VideoPresentation current = new VideoPresentation("key", "url");

    @Nullable
    public VideoPresentation getCurrentPresentation() {
        current.state = currentState;
        presentations.add(current);
        for (VideoPresentation information : presentations) {
            if (!PresentationState.STOP.equals(information.state))
                return information.clone();
        }
        return null;
    }

    public boolean startHasRun = false;

    @org.jetbrains.annotations.Nullable
    public String startArgs;

    @NonNull
    public Promise<VideoPresentation> start(@NonNull final String url) {
        startHasRun = true;
        startArgs = url;
        currentState = PresentationState.PLAY;
        current.url = url;
        presentations.add(current);
        return new Promise<VideoPresentation>(solver -> {
            VideoPresentation result = null;
            for (VideoPresentation presentation : presentations) {
                if (presentation.url.equals(url)) {
                    result = presentation;
                    break;
                }
            }
            solver.resolve(result);
        });
    }

    public boolean stopHasRun = false;
    @NonNull
    public Promise<VideoPresentation> stop() {
        stopHasRun = true;
        currentState = PresentationState.STOP;
        return Promise.resolve(null);
    }

    public boolean playHasRun = false;
    @NonNull
    public Promise<VideoPresentation> play() {
        playHasRun = true;
        currentState = PresentationState.PLAY;
        return Promise.resolve(null);
    }

    public boolean pauseHasRun = false;
    public long pauseArgs;
    @NonNull
    public Promise<VideoPresentation> pause(final long timestamp) {
        pauseHasRun = true;
        pauseArgs = timestamp;
        currentState = PresentationState.PAUSED;
        return Promise.resolve(null);
    }

    public boolean seekHasRun = false;
    public long seekArgs;
    @NonNull
    public Promise<VideoPresentation> seek(final long timestamp) {
        seekHasRun = true;
        seekArgs = timestamp;
        return Promise.resolve(null);
    }
}

File: test_app/integration_tests/ios_mocks/VoxeetSDKMock/Classes/VideoPresentationService.swift
import Foundation
import WebRTC

@objcMembers public class VideoPresentationService: NSObject {
    
    public internal(set) var current: VTVideoPresentation?
    public internal(set) var state: VTVideoPresentationState = .stopped
    
    var startHasRun: Bool = false
    var startArgs: URL?
    var startReturn: NSError?
    public func start(url: URL, completion: ((_ error: NSError?) -> Void)? = nil) {
        state = .playing
        startArgs = url
        startHasRun = true
        completion?(startReturn)
    }
    
    var stopHasRun: Bool = false
    var stopReturn: NSError?
    public func stop(completion: ((_ error: NSError?) -> Void)? = nil) {
        state = .stopped
        stopHasRun = true
        completion?(stopReturn)
    }
    
    var playHasRun: Bool = false
    var playReturn: NSError?
    public func play(completion: ((_ error: NSError?) -> Void)? = nil) {
        state = .playing
        playHasRun = true
        completion?(playReturn)
    }
    
    var pauseHasRun: Bool = false
    var pauseArgs: Int?
    var pauseReturn: NSError?
    public func pause(timestamp: Int, completion: ((_ error: NSError?) -> Void)? = nil) {
        state = .paused
        pauseArgs = timestamp
        pauseHasRun = true
        completion?(pauseReturn)
    }
    
    var seekHasRun: Bool = false
    var seekArgs: Int?
    var seekReturn: NSError?
    public func seek(timestamp: Int, completion: ((_ error: NSError?) -> Void)? = nil) {
        seekArgs = timestamp
        seekHasRun = true
        completion?(seekReturn)
    }
}

@objc public protocol VTVideoPresentationDelegate {
    /// Emitted when a video presentation is started.
    @objc func started(videoPresentation: VTVideoPresentation)
    /// Emitted when a video presentation is stopped.
    @objc func stopped(videoPresentation: VTVideoPresentation)
    /// Emitted when a video presentation is played.
    @objc func played(videoPresentation: VTVideoPresentation)
    /// Emitted when a video presentation is paused.
    @objc func paused(videoPresentation: VTVideoPresentation)
    /// Emitted when a video presentation is sought.
    @objc func sought(videoPresentation: VTVideoPresentation)
}
